### PR TITLE
Remove selection fix

### PR DESCRIPTION
**Pull request by deeptailor**
Update flex layout to use remove action, which includes a new function to remove domainObject from parent and update selection to reflect parent

### CODE REVIEW

File: src/plugins/remove/RemoveAction.js [hunk 1]
```diff
@@ -87,6 +88,14 @@ export default class RemoveAction {
         this.openmct.objects.mutate(parent, 'composition', composition);
     }
 
+    removeFromCompositionAndSetSelection(parent, child) {
```
Comment (akhenry) [inline src/plugins/remove/RemoveAction.js]: If the remove action itself doesn't use this function, it probably shouldn't be on the RemoveAction class. It's behavior that's only needed from flex layouts right now, so it should probably be there.

**Comment by akhenry** [comment]
Looks good. 

I am wondering if perhaps we should just call `.invoke()` on the RemoveAction? The only difference is that it would show a confirmation dialog, which we should probably have anyway since we're deleting an object. We would also need to add a check to the navigation logic so that it doesn't attempt to navigate if the parent object is the navigated object already.

**Comment by deeptailor** [comment]
@akherny - The reason I didn't want to call invoke, is because there are cases where a user deletes an entire column, and that would cause blocking messages for every domainObject in that column, will become very annoying

**Comment by deeptailor** [comment]
@akhenry - made some changes and additions.

### Changes
 - Moved set selection to flex layout, and removed the new function from remove action

### Additions 
- Updated the inspector to only update the views if the selection context has changed ie it was updating the views every time the selection fires a change event
- Force digest on the plot inspector view, to avoid delay between the asynchronous selection change and the angular world

File: src/ui/layout/BrowseBar.vue [hunk 2]
```diff
@@ -213,7 +215,11 @@ const PLACEHOLDER_OBJECT = {};
             }
         },
         mounted: function () {
-            this.notebookSnapshot = new NotebookSnapshot(this.openmct);
+
+            if (this.openmct.types.get('notebook')) {
```
Comment (akhenry) [inline src/ui/layout/BrowseBar.vue]: Nice way of checking if Notebook is installed

**Comment by akhenry** [comment]
Looks good to me!

**Pull request merged**